context/src/source.rs

```diff
@@ -7,6 +6,28 @@ pub trait SourceSegmentHolder {
     fn segment(&self) -> SourceSegment;
 }
 
+pub struct StaticSegmentHolder {
```

Comment:
This seems to add a lot of boilerplate. Simply add a constructor to `Observation` that take a raw segment. This type is public, and exposes an API that should be reserved to tests.